### ISSUE TITLE
core-console-backend-service pod fails with OOM errors

### ISSUE THREAD

**Issue by abbi-gaurav** [kyma-project/kyma#6939]
**Description**

On an around 50 days old Kyma cluster, the `core-console-backend-*` pod started failing continuously with OOM errors. Only after increasing the limit to 256M, I was able to get it up and running. The impacted the console UI.

**Expected result**
core-console-backend-service pods run without any issue.

**Actual result**

Pod was crashing with OOM and console UI was not navigatable.

**Comment by kwiatekus** [comment]
Hi @abbi-gaurav.

We identified that the default memory setup is not enough.
CBS have memory peaks whenever it experiences sync issues with API server.
We improved it already on master : https://github.com/kyma-project/kyma/pull/6904
 and we will define production grade profile for the component: https://github.com/kyma-project/kyma/issues/6679

**Comment by kwiatekus** [comment]
We have identified the root causes of the OOM problems of console-backend-service (CBS):

* Most of the time CBS keeps it memory usage below 64 MB. All the queries utilise its internal cache which is memory efficient. We realise that it is enough to set [requested memory](https://github.com/kyma-project/kyma/pull/6904/files#diff-58c94eeebeed6b0078bbf7b8dcae7ce4R43) at the level of `64 Mi`
* **There are peaks of memory consumption**. They occur when CBS is synchronising its cache with API server as  involves JSON objects parsing (memory expensive). They happen at the very beginning when CBS is started and periodically, according to the re-sync policy (every 10 minutes).
Peak values often exceeded the limit, therefore we **[increased the limit](https://github.com/kyma-project/kyma/pull/6904/files#diff-58c94eeebeed6b0078bbf7b8dcae7ce4R41)** to `300 Mi`
* CBS is divided into several domains and each domain use its own http client to sync data with API  server. We adopted the cache re-sync policy so that re-sync operations for different domains are not happening in the same time - therefore, **higher memory consumption at the time of re-sync is more distributed**. Additionally, we have limited the default API burst rate - limiting the number of simultaneous objects to parse. (https://github.com/kyma-project/kyma/pull/6887)
* We have identified and **fixed memory leak** - unclosed connection upon authentication error (https://github.com/kyma-project/kyma/pull/6870)

**Comment by sjanota** [comment]
We are still experiencing problems with CBS stability. Let's reopen the thread.

**Comment by sjanota** [comment]
Some observations:
- When run in verbose mode k8s client does not log any errors upon OOMKill. It looks either it is not related to network errors as we suspected or it happens so fast that logger can't handle the messages.
- In namespace subscription, applications are resolved per subscription, not per event. It means that we potentially do a lot of unnecessary downloading and for sure unnecessary processing.

**Comment by sjanota** [comment]
Something is clearly wrong with the Namespaces view. 

<img width="1095" alt="Screen Shot 2020-03-20 at 12 07 34" src="https://user-images.githubusercontent.com/5119235/77158350-6da58900-6aa3-11ea-8b5a-045fb153345b.png">

I opened one tab with the console on Namespaces View. Then around 11:55, I changed the view to pods in kyma-system. Also "steps" are more or less synchronized with cache resyncs that generate a lot of events.
